objectstore-server/src/authentication.rs

```diff
@@ -0,0 +1,94 @@
+use axum::extract::FromRequestParts;
+use axum::http::request::Parts;
+use axum::http::{StatusCode, header};
+use jsonwebtoken::errors::Result as JwtResult;
+use jsonwebtoken::{DecodingKey, Validation, decode};
+use objectstore_service::{Scope, Usecase};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+pub struct Claim {
+    pub usecase: Usecase,
+    pub scope: Scope,
+}
+
+pub fn decode_auth_header(token: &str) -> JwtResult<Claim> {
+    let key = DecodingKey::from_secret(b"TODO");
```

Comment:
Let's load the key from config? We can then define it using environment variables.